lib/pavilion/plugins/commands/status.py

```diff
@@ -139,6 +112,57 @@ def get_tests(pav_cfg, args, errfile):
     return test_list
 
 
+def trim_tests(pav_cfg, args, errfile):
+    """Consolidates and filters test_status list based on
+    the limiting argument flags under 'pav status'. """
+
+    test_statuses = []
+    test_obj_list = []
+
+    if args.all and not args.limit:
+        tests = test_run.get_latest_tests(pav_cfg)
+    else:
+        tests = test_run.get_latest_tests(pav_cfg, args.limit)
```

Comment:
You're limiting test by time twice. Once by get_latest_tests standards, and once by your standards. The first filtering may return different tests than you would expect by your standards. 

According to pycharm, the only usage of get_latest_tests is in the status command. 
Here's what I think you should do: 
  - Get rid of the 'get_latest_tests' function (entirely)
  - Use dir_db.select instead. 

You'll need to dynamically generate a filter function. That will be educational.